docs/developer/releasing.md

```diff
@@ -113,25 +124,55 @@ For patch releases, the release shepard will:
 
 3. Create the patch release tag.
 
-After the release shepard pushes a new tag, they must [publish the release](#publishing-the-release).
+4. [Publish the release](#publishing-the-release)
 
 ### Updating code
 
-The codebase must be updated whenever a new release is being made to reference
-the upcoming release tag:
+The codebase must be updated whenever a new release is being made to reference the upcoming release tag.
 
-* Modify `CHANGELOG.md` with a new version number and its release date.
-* Go through the entire repository and find references to the previous release
-  version, updating them to reference the new version.
-* Run `make generate-manifests` and `make generate-dashboards` to update
-  manifests in case they are stale.
+#### Update the changelog
 
 NOTE: Any time CHANGELOG.md is updated for a release, it should first be done
 via PR to the release branch, and then by a second PR to main.
 
+When creating a release **candidate** (an `-rc.N` version):
+
+* Add a new section in the changelog for the new release candidate and include 
+today's date. E.g. "v0.31.0-rc.0 (2023-01-26)".
+* All items form the "unreleased" section will move to a new section for the upcoming release version.
+* See [here](https://github.com/grafana/agent/pull/2838/files) for an example pull request.
+* Sanity check that the changelog doesn't contain features which don't belong to it. Sometimes features get added to the wrong version in the changelog due to bad merges.
+
+When creating a **stable** release:
+
+* Replace the version from the release candidate section to be the stable release instead 
+of the `-rc.N` version.
+* We do not leave release candidates in the changelog once there is a stable version for them.
+* Make sure to also update the **date** in the title.
+* See [here](https://github.com/grafana/agent/pull/2873/files) for an example pull request.
+
+#### Search and replace the old version with the new one
```

Comment:
Can we add a notice here to not touch the `operations/helm` directory? That should be updated independently from releases for now.

Reply:
Will do, thanks!